journal2cri/journal2cri/journal2cri.go

```diff
@@ -0,0 +1,104 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package journal2cri contains functions to convert a systemd journal to a CRI logfile
+package journal2cri
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/coreos/go-systemd/sdjournal"
+)
+
+// ProcessEntry parses a journal entry into a struct containing the info needed for CRI logging
+func ProcessEntry(entry *sdjournal.JournalEntry) *CRIEntry {
+	rktAppName, ok := entry.Fields["SYSLOG_IDENTIFIER"]
+	if !ok {
+		// Not an app entry
+		return nil
+	}
+
+	// App names have the special format of 'appname-instanceNumber' by convention
+	parts := strings.SplitAfterN(rktAppName, "-", 2)
+	if len(parts) != 2 {
+		return nil
+	}
+
+	appName := parts[0]
+	appNumberStr := parts[1]
+
+	appNumber, err := strconv.Atoi(appNumberStr)
+	if err != nil {
+		return nil
+	}
+	appName = strings.TrimRight(appName, "-")
+	if len(appName) == 0 {
+		return nil
+	}
+
+	outStream := entry.Fields["_TRANSPORT"]
+	if outStream != string(CRIStreamStdout) && outStream != string(CRIStreamStderr) {
```

Comment:
Thanks, I wasn't aware and didn't do any testing there. I guess there's no way to get that then?